### PR TITLE
Redesigned fee structure

### PR DESCRIPTION
**Pull request by Mingela**

### CODE REVIEW

File: data-collector/src/main/kotlin/com/d3/datacollector/repository/BillingRepository.kt [hunk 1]
```diff
@@ -7,14 +7,22 @@ package com.d3.datacollector.repository
...
 import org.springframework.data.repository.CrudRepository
+import java.math.BigDecimal
 import java.util.*
 
 interface BillingRepository : CrudRepository<Billing, Long> {
 
-    @Query("SELECT b FROM Billing b WHERE b.domainName = ?1 and b.asset = ?2 and b.billingType = ?3")
-    fun selectByDomainBillingTypeAndAsset(
+    @Query("SELECT b FROM Billing b " +
+            "WHERE b.domainName = ?1 and b.asset = ?2 and b.billingType = ?3" +
```
Comment (dolgopolovwork) [inline data-collector/src/main/kotlin/com/d3/datacollector/repository/BillingRepository.kt]: I would use param names instead of numbers.

Comment (Mingela) [reply data-collector/src/main/kotlin/com/d3/datacollector/repository/BillingRepository.kt]: fixed 😄

File: data-collector/src/main/kotlin/com/d3/datacollector/model/BillingModel.kt [hunk 2]
```diff
@@ -65,4 +88,22 @@ data class Billing(
     protected fun onUpdate() {
         updated = Date().time
     }
+
+    override fun hashCode() =
+        (feeDescription + domainName + billingType + asset + destination +
+                minAmount.toPlainString() + maxAmount.toPlainString()).hashCode()
+
+    override fun equals(other: Any?): Boolean {
```
Comment (yurivin) [inline data-collector/src/main/kotlin/com/d3/datacollector/model/BillingModel.kt:99]: What about the null check at the beginning of the method? If 'other' is null, then we can return false at the beginning.

**Review by yurivin — APPROVED** [review]

File: data-collector/src/test/kotlin/com/d3/datacollector/tests/ControllersTest.kt [hunk 3]
```diff
@@ -122,6 +131,23 @@ class ControllersTest : TestEnv() {
...
+        feeFraction = feeFraction,
+        feeComputation = feeComputation,
+        feeAccount = feeAccount,
+        minAmount = minAmount,
+        maxAmount = maxAmount,
+        minFee = minFee,
+        maxFee = maxFee,
+        created = created,
+        updated = updated
+    )
```
Comment (Alexey-N-Chernyshov) [inline data-collector/src/test/kotlin/com/d3/datacollector/tests/ControllersTest.kt]: eol

**Review by Alexey-N-Chernyshov — APPROVED** [review]

**Review by yurivin — APPROVED** [review]

**Review by AntonKtrue — APPROVED** [review]

**Comment by Mingela** [comment]
Done in another repo

**Pull request closed without merging**